src/Coordination/KeeperStateMachine.cpp

```diff
@@ -383,7 +383,13 @@ void KeeperStateMachine::create_snapshot(nuraft::snapshot & s, nuraft::async_res
     };
 
 
-    LOG_DEBUG(log, "In memory snapshot {} created, queueing task to flash to disk", s.get_last_log_idx());
+    if (keeper_context->server_state == KeeperContext::Phase::SHUTDOWN)
+    {
+        snapshot_task.create_snapshot(std::move(snapshot_task.snapshot));
```

Comment:
Maybe add logging about it, something simple like "Creating snapshot because server is shutting down and `create_snapshot_on_exit` is enabled".